rs/state_machine_tests/src/lib.rs

```diff
@@ -2183,7 +2189,11 @@ impl StateMachine {
                 dk.sign_message(&context.schnorr_args().message).to_vec()
             }
             _ => {
-                panic!("No Schnorr key with specified key id found");
+                debug!(
+                    self.replica_logger,
+                    "No Schnorr key with specified key id found"
+                );
+                return None;
```

Comment:
What would the mainnet do in such a case?

Reply:
@eichhorl Would you know? We discussed this in the crypto team meeting and thought that this may be in the realm of consensus.

Reply:
If the key doesn't exist at all we [return](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/execution_environment/src/execution_environment.rs?L1186-1189) this [error response](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/execution_environment/src/execution_environment.rs?L3714-3720). If it exists but signing is disabled we return a different [error response](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/execution_environment/src/execution_environment.rs?L2627-2633).

Reply:
Thanks for the details! In my opinion, we should the same here if we don't panic.

Reply:
I changed the code to return [the error response](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/execution_environment/src/execution_environment.rs?L3714-3720) mentioned by @eichhorl for situations where the key doesn't exist.